### PR TITLE
Caching --requires testing 

### PR DESCRIPTION
**Pull request by mrhut10**
observational data

- zeit deployments looks like a decreased response times about 200 ms
- on local deployments more like a 500 to 600 ms decrease in response times
- and when under load will greatly reduce database calls

Testing still required by @Irev-Dev 
- do flashes still work where they are needed ???
- does the GET /reg url still work???

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,9 +6,9 @@
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
     "watch-server": "nodemon ./server/start.js --ignore public/",
-    "build-js": "parcel build -d server/client/dist server/client/client.js --public-url ./",
-    "build-css": "concurrently \"parcel build -d server/client/dist server/client/normalise.css --public-url ./\" \"parcel build -d server/client/dist server/client/style.css --public-url ./ \" ",
-    "build-html": "parcel build -d server/client/dist server/client/main.html --public-url ./",
+    "build-js": "parcel build server/client/client.js --out-dir server/client/dist  --public-url ./ --log-level 1",
```
Comment (mrhut10) [inline package.json:9]: this reduces the logs being filled with build statements, will only log errors now

File: package.json [hunk 2]
```diff
@@ -66,6 +67,6 @@
     "eslint": "^4.16.0",
     "eslint-config-airbnb": "^16.1.0",
     "eslint-plugin-import": "^2.8.0",
-    "parcel-bundler": "^1.5.0"
+    "parcel-bundler": "^1.7.1"
```
Comment (mrhut10) [inline package.json:70]: updated parcel version to allow for the --log-level feature

File: server/controllers/userController.js [hunk 3]
```diff
@@ -110,21 +110,30 @@ exports.subscriptionChange = async (req, res) => {
   res.redirect('/');
 };
 
+function uncachedCountdown() {
+  return User.count();
+}
+
+/* removed this block of code as we are caching the full server responses now higher up
 function CachedCountDown() {
```
Comment (mrhut10) [inline server/controllers/userController.js:118]: this function could likely be completely removed or at least moved into a way where we write a generalised caching function within the caching library.

File: server/server.js [hunk 4]
```diff
@@ -62,11 +63,16 @@ app.use((req, res, next) => {
   next();
 });
 
-app.get('/', async (req, res) => {
-  res.render('main', { countDown: 300 - await userController.cachedCountDown.getCount() });
-});
+// caching the main page's response
+app.get(
+  '/',
+  responseCache(20 * 1000), // cache body data for 20 seconds
```
Comment (mrhut10) [inline server/server.js:69]: these caching times are arbitory for when i was testing it. please review these times @Irev-Dev

**Review by mrhut10 — COMMENTED** [review]
@Irev-Dev 
- does flashes still work?
- does GET /reg url still work
- review caching times for routes
- have a fun time

**Review by Irev-Dev — CHANGES_REQUESTED** [review]
flashes are in-fact now broken, they get bundled up as part of the cache eg when you sign up you are given a verify link that when navigated to gives you a success flash, but if you navigate to it for a second time you should get an info flash saying that you have already verified, but now with these changes you keep getting the success flash until the cache refreshes.

**Comment by mrhut10** [comment]
OK I should be able to fix that fairly easily if you haven't already, I'll close the request now and resubmit a fix. 
:)
Kind regards

**Pull request closed without merging**